File: atomic-x/android/src/main/kotlin/io/trtc/tuikit/atomicx/audiorecorder/AudioRecorderPlugin.kt
package io.trtc.tuikit.atomicx.audiorecorder

import android.app.Activity
import android.util.Log
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.MethodChannel

class AudioRecorderPlugin(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    
    companion object {
        private const val TAG = "AudioRecorderPlugin"
        private const val METHOD_CHANNEL_NAME = "atomic_x/audio_recorder"
        private const val EVENT_CHANNEL_NAME = "atomic_x/audio_recorder_events"
    }

    private val methodChannel: MethodChannel = MethodChannel(
        flutterPluginBinding.binaryMessenger,
        METHOD_CHANNEL_NAME
    )
    
    private val eventChannel: EventChannel = EventChannel(
        flutterPluginBinding.binaryMessenger,
        EVENT_CHANNEL_NAME
    )
    
    private var handler: AudioRecorderHandler? = null

    init {
        // Note: Activity will be set when available through ActivityAware
        Log.d(TAG, "AudioRecorderPlugin initialized")
    }

    fun attachToActivity(activity: Activity) {
        handler = AudioRecorderHandler(activity, methodChannel, eventChannel)
        methodChannel.setMethodCallHandler(handler)
        eventChannel.setStreamHandler(handler)
        Log.d(TAG, "AudioRecorderPlugin attached to activity")
    }

    fun detachFromActivity() {
        handler?.dispose()
        handler = null
        methodChannel.setMethodCallHandler(null)
        eventChannel.setStreamHandler(null)
        Log.d(TAG, "AudioRecorderPlugin detached from activity")
    }

    fun dispose() {
        detachFromActivity()
    }
}

File: atomic-x/ios/Classes/AudioRecorder/AudioRecorderPlugin.swift
import Flutter
import UIKit

public class AudioRecorderPlugin: NSObject {
    private var methodChannel: FlutterMethodChannel?
    private var eventChannel: FlutterEventChannel?
    private var handler: AudioRecorderHandler?
    
    init(registrar: FlutterPluginRegistrar) {
        super.init()
        
        methodChannel = FlutterMethodChannel(
            name: "atomic_x/audio_recorder",
            binaryMessenger: registrar.messenger()
        )
        
        eventChannel = FlutterEventChannel(
            name: "atomic_x/audio_recorder_events",
            binaryMessenger: registrar.messenger()
        )
        
        handler = AudioRecorderHandler(
            methodChannel: methodChannel!,
            eventChannel: eventChannel!
        )
        
        methodChannel?.setMethodCallHandler { [weak self] (call, result) in
            self?.handler?.handle(call, result: result)
        }
        
        eventChannel?.setStreamHandler(handler)
    }
    
    public func dispose() {
        handler?.dispose()
        handler = nil
        methodChannel?.setMethodCallHandler(nil)
        eventChannel?.setStreamHandler(nil)
        methodChannel = nil
        eventChannel = nil
    }
}
